src/NuGet.Clients/NuGet.VisualStudio.Implementation/VsTemplateWizard.cs

```diff
@@ -66,11 +66,22 @@ IVsProjectAdapterProvider vsProjectAdapterProvider
             _sourceProvider = sourceProvider;
             _vsProjectAdapterProvider = vsProjectAdapterProvider;
 
-            _preinstalledPackageInstaller = new PreinstalledPackageInstaller(_packageServices, _solutionManager, _settings, _sourceProvider, (VsPackageInstaller)_installer, _vsProjectAdapterProvider);
-
             PumpingJTF = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory);
         }
 
+        private PreinstalledPackageInstaller PreinstalledPackageInstaller
+        {
+            get
+            {
+                if (_preinstalledPackageInstaller == null)
```

Comment:
nit: Consider a lazy to avoid a race condition.

Reply:
thx. moved to Lazy<>... good practice, that I haven't done much yet.